test/integration/narrow/routes/farmer-apply/sheep.test.js

```diff
@@ -0,0 +1,42 @@
+
+const getCrumbs = require('../../../../utils/get-crumbs')
+
+const auth = { credentials: { reference: '1111', sbi: '111111111' }, strategy: 'basic' }
+
+describe('Sheep test', () => {
+  test('GET /farmer-apply/sheep route returns 200', async () => {
+    const options = {
+      method: 'GET',
+      url: '/farmer-apply/sheep',
+      auth
+    }
+    const res = await global.__SERVER__.inject(options)
+    expect(res.statusCode).toBe(200)
+  })
+  test('POST /farmer-apply/sheep route returns 302', async () => {
+    const crumb = await getCrumbs(global.__SERVER__)
+    const options = {
+      method: 'POST',
+      url: '/farmer-apply/sheep',
+      payload: { crumb, sheep: 'yes' },
+      auth,
+      headers: { cookie: `crumb=${crumb}` }
+    }
+    const res = await global.__SERVER__.inject(options)
+    expect(res.statusCode).toBe(302)
+    expect(res.headers.location).toEqual('/farmer-apply/pigs')
+  })
+  test('POST /farmer-apply/sheep route returns Error', async () => {
+    const crumb = await getCrumbs(global.__SERVER__)
+    const options = {
+      method: 'POST',
+      url: '/farmer-apply/sheep',
+      payload: { crumb, sheep: null },
+      auth,
+      headers: { cookie: `crumb=${crumb}` }
+    }
+    const res = await global.__SERVER__.inject(options)
+    expect(res.payload).toContain('Select yes if you keep more than 20 sheep')
```

Comment:
See the comment regarding using `cheerio` to enable specific content to be selected from the page.